File: src/main/scala/game/entity/character/RelationshipType.scala
package game.entity.character

trait RelationshipType{
    val name:String
}

object RelationshipTypes{
    
    case object ParentOf extends RelationshipType{
        override val name = "parent"
    }
    case object ChildOf extends RelationshipType{
        override val name = "child"
    }
    case object SiblingsWith extends RelationshipType{
        override val name = "sibling"
    }
    case object CousinsWith extends RelationshipType{
        override val name = "cousin"
    }
    case object GrandParentOf extends RelationshipType{
        override val name = "grandparent"
    }
    
    case object NotRelated extends RelationshipType{
        override val name = "not related to"
    }
    case object AcquaintencesWith extends RelationshipType{
        override val name = "acquaintence"
    }
    case object FriendsWith extends RelationshipType{
        override val name = "friend"
    }
    case object InLoveWith extends RelationshipType{
        override val name = "in love"
    }
    case object EnemiesWith extends RelationshipType{
        override val name = "enemy"
    }
    
}

File: src/main/scala/game/entity/character/RelationshipType.java
package game.entity.character;

public enum RelationshipType {
	
	MarriedTo("spouse"),
	ParentOf("child"),
	ChildOf("parent"),
	SiblingsWith("sibling"),
	AcquaintencesWith("acquaintence"),
	FriendsWith("friend"),
	InLoveWith("love interest"),
	EnemiesWith("enemy")
	;
	
	private String objectOfRelationship;
	
	private RelationshipType(String objectOfRelationship){
		this.objectOfRelationship = objectOfRelationship;
	}

	public String getObjectOfRelationship() {
		return objectOfRelationship;
	}

	
	public static final int MIN_VAL = -255;
	public static final int MAX_VAL = 255;
}
